src/blog/2024/03/using-kafka-with-node-red.md

```diff
@@ -0,0 +1,143 @@
+
+<!--more-->
+
+During my recent visit to the Node-RED Forum, I noticed that while some Kafka-related queries have been answered nicely, others remain unanswered or have not been satisfactorily addressed, leaving users feeling stuck. To address this issue, we've created a comprehensive Kafka guide covering everything you need to know about Kafka, from installation and connection to data transmission. For newcomers to Kafka, we recommend reading our previous blog on [how Kafka is applied in manufacturing](https://flowfuse.com/blog/2024/03/using-kafka-in-manufacturing/), where we've covered the basics and practical applications extensively.
+
+## Discussing problem and potential solution
+Let's start by discussing a problem: imagine a temperature sensor network across a city. We need to centralize and analyze this data in real time for effective monitoring and visualization.
+
+To resolve this problem we will use Kafka, Temperature sensors will feed data into Kafka through the Kafka producer. To retrieve real-time data for visualization and monitoring, we’ll be using Kafka Consumer. We will organize the data by region. The temperature data for each region will be managed in a specific Kafka topic partition.
+
+While in this guide, we will generate simulated data using random number expression and run both producers and consumers on the same system, practical scenarios often involve distributed setups across different devices or systems.
+
+## Installing and running Kafka locally 
+In this part, we’ll be installing Kafka locally using Docker to simplify the installation process, so make sure you’ve got Docker installed before you dive in.
+
+1. Pull the zookeeper image if it is not already, and run the zookeeper container.
+```
+docker run -p 2181:2181 zookeeper
+
+```
+
+2. Pull the Kafka image if it is not already, and run Kafka Container, expose PORT 9092.
+```
+docker run -p 9092:9092 `
+-e KAFKA_ZOOKEEPER_CONNECT=<Your_Private_Ip>.1:2181 `
+-e KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://<Your_Private_Ip>:9092 `
+-e KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1 `
+confluentinc/cp-kafka
+```
+
+## Running Kafka on the cloud
+To run Kafka on the cloud, you can consider utilizing any cloud service according to your preferences. For a guide on running Kafka on a cloud platform, the procedures may differ. You can refer to the documentation provided by your preferred cloud service for detailed instructions.
+
+During the writing of this tutorial, I utilized [Aiven’s cloud data platform](https://aiven.io/kafka-connect) which offers the option to use Kafka in the free trial. However, you are free to choose any cloud service that suits your requirements and preferences.
+
+## Installing Dashboard 2.0
```

Comment:
Since @joepavitt wrote an amazing guide, https://flowfuse.com/blog/2024/03/dashboard-getting-started/, I think we should start linking there instead.

Something along the lines of `[If you're new to Dashboard 2.0, follow our guide](https://flowfuse.com/blog/2024/03/dashboard-getting-started/)` should now suffice